### PR TITLE
Refactor dropdown menu for power actions (#bnc955810)

### PR DESCRIPTION
**Pull request by vwallfahrer**
- Dropdown menu collapse after action
- Shows flash message
- Shows blockUI while rebooting

### CODE REVIEW

**Comment by vuntz** [comment]
Why do we need the blockui for this?

**Comment by vwallfahrer** [comment]
@vuntz  It takes a few seconds until the page is reloaded and node starts to reboot.

File: crowbar_framework/app/controllers/nodes_controller.rb [hunk 1]
```diff
@@ -307,19 +307,42 @@ def hit
...
+        else
+          case action
+          when "reinstall", "reset", "update", "delete"
+            machine.set_state(action)
+          when "reboot", "shutdown", "poweron", "powercycle", "poweroff", "identify", "allocate"
+            machine.send(action)
+          else
+            render json: { error: "Invalid hit request '#{action}'" }, status: 500 && return
+          end
+            head :ok
```
Comment (tboerger) [inline crowbar_framework/app/controllers/nodes_controller.rb]: As hound say, too much indention :)

File: crowbar_framework/app/controllers/nodes_controller.rb [hunk 2]
```diff
@@ -307,19 +307,42 @@ def hit
...
+        end
+      end
+
+      format.html do
+        if machine.nil?
+          flash[:alert] = "Could not find node '#{name}'"
+          redirect_to nodes_path, status: 404
```
Comment (tboerger) [inline crowbar_framework/app/controllers/nodes_controller.rb]: I think that should be `dashboard_url`. Hint: Always use the `_url` helpers for redirects.

**Comment by tboerger** [comment]
In general I would just drop the `remote: true` flags from the links to make it real requests instead of AJAX.

**Comment by vuntz** [comment]
Travis failed, can you check it out?

**Comment by vwallfahrer** [comment]
@vuntz  yes, I will do it

**Comment by vuntz** [comment]
@vwallfahrer can you maybe squash the commits?

File: crowbar_framework/app/assets/javascripts/application.js [hunk 3]
```diff
@@ -95,9 +78,35 @@ jQuery(document).ready(function($) {
     ]
   });
 
-  setBlockUI('submit', $('[data-blockui]'));
-  setBlockUI('click', $('body.backups').find('[data-blockui]'));
-  setBlockUI('click', $('body.installer-upgrades').find('[data-blockui]'));
```
Comment (vuntz) [inline crowbar_framework/app/assets/javascripts/application.js]: Aren't we breaking the blockui bit for backup/installer here?

File: crowbar_framework/config/locales/crowbar/en.yml [hunk 4]
```diff
@@ -442,6 +442,7 @@ en:
       powercycle: 'Power Cycle'
       poweroff: 'Power Off'
       identify: 'Identify'
+      blockui_message: 'Rebooting nodes...'
```
Comment (jsuchome) [inline crowbar_framework/config/locales/crowbar/en.yml]: Is it nodes, or just one node? I believe Power action is only available from 1 node view....

**Comment by jsuchome** [comment]
It seems to work well - block message is shown while rebooting

File: crowbar_framework/config/locales/crowbar/en.yml (hunk 4 above)
Comment (vwallfahrer) [inline crowbar_framework/config/locales/crowbar/en.yml]: @jsuchome  you are right. I will change that

File: crowbar_framework/app/assets/javascripts/application.js (hunk 3 above)
Comment (MaximilianMeister) [inline crowbar_framework/app/assets/javascripts/application.js]: yes, this needs to be adapted https://github.com/crowbar/crowbar-core/blob/master/crowbar_framework/app/helpers/backups_helper.rb#L42

**Comment by jsuchome** [comment]
I can't judge javascript changes, but as for functionality, LGTM

File: crowbar_framework/app/assets/javascripts/application.js (hunk 3 above)
Comment (vuntz) [inline crowbar_framework/app/assets/javascripts/application.js]: I see the backup part was done, but not the installer part.

**Comment by tboerger** [comment]
After fixing all blockui parts within the whole application you get my :+1:

**Pull request closed without merging**